src/test/java/nl/tudelft/ewi/devhub/server/database/controllers/CommitsTest.java

```diff
@@ -101,6 +108,17 @@ public void testPersistWithParent() {
 		Commit b = createCommit(group.getRepository());
 		assertThat(b.getParents(), Matchers.contains(a));
 	}
+
+	@Test
+	public void testCommitLineChanges() {
+
+		Commit commit = createCommit(group.getRepository());
+		assertTrue(commit.getLinesAdded() == 0);
+		commit.setLinesAdded(69);
+		commit.setLinesRemoved(666);
+		assertTrue(commit.getLinesAdded() == 69);
+		assertTrue(commit.getLinesRemoved() == 666);
```

Comment:
These tests verify the integration with Hibernate. There are a couple of things that can go wrong with mapping the value:

1) The field is not properly mapped or not present in the `changelog.xml`: the test will not run at all and fail during the `EntityManagerFactory` initialization. This does not to be tested if there is any test in the test codebase that boots up an `EntityManagerFactory` (which all of these controller integration tests do).

2) The field is not mapped at all. In order to test these, one should override the default value (most likely `null`), flush the changes (so they are written to the database) and refresh the entity (so that the value from the database overrides the in-memory value if different) and assert that the value equals the expected value.

3) The field has unreasonable constaints on it. For example a foreign-key constraint to the wrong entity, or a size that is too small to fit its regular content. By using sane values for the test for 2, this can easily be covered in that test.

 In order to make the test complete, you should flush the changes to the database ([`entityManager.flush()`](http://docs.oracle.com/javaee/7/api/javax/persistence/EntityManager.html#flush--)), then refresh the entity ([`entityManager.refresh(commit)`](http://docs.oracle.com/javaee/7/api/javax/persistence/EntityManager.html#refresh-java.lang.Object-)) and finally assert that the values are correct.